src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String query = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query);) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = resultSetToBook(resultSet);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a connection to the DB", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        Long id = book.getId();
+        Optional<Book> optionalBook = findById(id);
+        if (!optionalBook.isPresent()) {
+            throw new RuntimeException("Book not found with id " + id);
+        }
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query);) {
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setLong(3, id);
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to update at less one row,"
+                        + " but inserted 0 rows.");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        int updatedRows = 0;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query);) {
+            statement.setLong(1, id);
+            updatedRows = statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return (updatedRows > 0);
+    }
+
+    private Book resultSetToBook(ResultSet resultSet) {
+        Book book = new Book();
+        try {
+            Long idBook = resultSet.getLong("id");
+            String titleBook = resultSet.getString("title");
+            BigDecimal priceBook = resultSet
+                    .getObject("price",BigDecimal.class);
+            book.setId(idBook);
+            book.setTitle(titleBook);
+            book.setPrice(priceBook);
+        } catch (SQLException e) {
+            throw new DataProcessingException("The query result was not "
+                    + "added to the book", e);
+        }
+        return book;
```

Comment:
no need to handle SQLException in this method, just throw up to methods that know how to catch it.